xapian-letor/ranker/listnet_ranker.cc

```diff
@@ -134,12 +135,16 @@ ListNETRanker::train(const std::vector<Xapian::FeatureVector> & training_data) {
 
     // iterations
     for (int iter_num = 1; iter_num <= iterations; ++iter_num) {
-	// initialize Probability distributions of y and z
-	prob_distrib_vector prob = init_probability(training_data, new_parameters);
-	// compute gradient
-	vector<double> gradient = calculate_gradient(training_data, prob);
-	// update parameters: w = w - gradient * learningRate
-	update_parameters(new_parameters, gradient, learning_rate);
+	for (size_t it = 0; it < training_data.size(); ++it) {
```

Comment:
Again, an opportunity to use C++11 range `for`.

Reply:
fixed in a fixup commit.